### ISSUE TITLE
4.4.7 broke array types in Context

### ISSUE THREAD

**Issue by dustinsmith1024** [honojs/hono#3131]
### What version of Hono are you using?

4.4.7+

### What runtime/platform is your app running on?

Cloudflare Workers

### What steps can reproduce the bug?

I am attempting to upgrade to the latest 4.4 versions and I get a types issue with Array based elements in the HonoContext.

Sample context:

```
export interface DC {
  name: string;
  url: string;
}

declare module 'hono' {
  interface ContextVariableMap {
    jwt: JWTInterface;
    dcs: DC[];
    allowAllOrigins: boolean;
    allowedOrigins: string[];
    appToken: string;
    appEnv: string;
  }
}
```

### What is the expected behavior?

No type errors should happen. This doesn't happen on 4.4.6, so maybe related to https://github.com/honojs/hono/pull/2987?

### What do you see instead?

```
➜ pnpm run -F gateway-router tsc

> router@0.1.0 tsc router
> tsc --noEmit                                 ~~

src/file.ts:327:16 - error TS7006: Parameter 'dc' implicitly has an 'any' type.

327       .filter((dc) => userDcs.includes(dc.name))
                   ~~

src/file.ts:328:13 - error TS7006: Parameter 'dc' implicitly has an 'any' type.

328       .map((dc) => {
                ~~

src/file.ts:50:40 - error TS7006: Parameter 'dc' implicitly has an 'any' type.

50     if (!body.dc || c.get('dcs').find((dc) => dc.name === body.dc) === undefined) {
                                          ~~

src/file2.ts:180:55 - error TS7006: Parameter 'allowed' implicitly has an 'any' type.

180         const allowed = c.get('allowedOrigins').find((allowed) => origin === allowed);
                                                          ~~~~~~~
```

**Comment by yusukebe** [comment]
Hi @dustinsmith1024 

Is this way of reproducing it correctly?

https://github.com/user-attachments/assets/52a1fe57-c3a1-434e-a64c-a8c6976ba9a0

If so, it does not throw errors. Can you provide a minimal project to reproduce it?

**Comment by dustinsmith1024** [comment]
Here you go: https://github.com/dustinsmith1024/hono-types-issue

**Comment by yusukebe** [comment]
@dustinsmith1024 Thanks. Will be fixed soon.

**Comment by dustinsmith1024** [comment]
Awesome thanks - Should I be doing something different with Context to prevent it?

**Comment by yusukebe** [comment]
@dustinsmith1024 

> Should I be doing something different with Context to prevent it?

It's a bug, so you don't have to add anything. Just wait for a new version!

**Comment by yusukebe** [comment]
Hi @dustinsmith1024 

This may be fixed in the latest version `v4.5.0`. Can you try it?

**Comment by yusukebe** [comment]
We can close this issue now. If you still have a problem, feel free to reopen.
